### DISCUSSION TITLE
مسدودسازی سایت‌ها و اپلیکیشن‌های ایرانی از سمت سرور

### DISCUSSION THREAD

**Discussion by SaintShit** [Gozargah/Marzban#38] [Ideas]
آپدیت: پست کامل تر یکی از دوستان برای [بستن سایت های ایرانی در مرزبان](https://telegra.ph/%D8%A8%D8%B3%D8%AA%D9%86-%D8%B3%D8%A7%DB%8C%D8%AA-%D9%87%D8%A7%DB%8C-%D8%A7%DB%8C%D8%B1%D8%A7%D9%86%DB%8C-%D8%AF%D8%B1-%D9%85%D8%B1%D8%B2%D8%A8%D8%A7%D9%86-03-13)

**Comment by ghost** [comment]
> آپدیت: پست کامل تر یکی از دوستان برای [بستن سایت های ایرانی در مرزبان](https://telegra.ph/%D8%A8%D8%B3%D8%AA%D9%86-%D8%B3%D8%A7%DB%8C%D8%AA-%D9%87%D8%A7%DB%8C-%D8%A7%DB%8C%D8%B1%D8%A7%D9%86%DB%8C-%D8%AF%D8%B1-%D9%85%D8%B1%D8%B2%D8%A8%D8%A7%D9%86-03-13)

سلام من این آموزش و بررسی کردم و متوجه یه سری اضافه کاری شدم برای نمونه category-ir شامل تمامی مواردی هست که خط به خط فراخوانی شدن و همچنین دسترسی به telegra.ph مشکل تره برای همین آموزش رو اینجا گام به گام مینویسم.
1. دستورات زیر رو تو محیط ترمینال سرورتون بزنید. این دستورات فایل های مورد نیاز رو دانلود میکنه.
```sh
mkdir -p /var/lib/marzban/assets/
```
```sh
wget -O /var/lib/marzban/assets/geosite.dat https://github.com/v2fly/domain-list-community/releases/latest/download/dlc.dat
```
```sh
wget -O /var/lib/marzban/assets/geoip.dat https://github.com/v2fly/geoip/releases/latest/download/geoip.dat
```
```sh
wget -O /var/lib/marzban/assets/iran.dat https://github.com/bootmortis/iran-hosted-domains/releases/latest/download/iran.dat
```
2. در پوشه marzban فایل env رو ویرایش کنید و مقدار زیر رو به اون اضافه کنید.
```env
XRAY_ASSETS_PATH = "/var/lib/marzban/assets/"
```
3. فایل xray_config.json رو باز کنید و بخش routing رو مطابق نمونه زیر پر کنید یا کلا بخش routing رو پاک کنید و مقدار زیر رو جایگزین کنید.
(توجه کنید که فقط بخش routing رو باید پاک کنید. اگر کلا تا انتهای فایل پاک کردین باید یدونه { به آخرش اضافه کنید. )
```json
    "routing": {
        "domainStrategy": "IPIfNonMatch",
        "rules": [
            {
                "type": "field",
                "outboundTag": "blackhole",
                "ip": [
                    "geoip:private",
                    "geoip:ir"
                ]
            },
            {
                "type": "field",
                "port": 53,
                "network": "tcp,udp",
                "outboundTag": "DNS-Internal"
            },
            {
                "type": "field",
                "outboundTag": "blackhole",
                "protocol": [
                    "bittorrent"
                ]
            },
            {
                "outboundTag": "blackhole",
                "domain": [
                    "regexp:.*\\.ir$",
                    "ext:iran.dat:ir",
                    "ext:iran.dat:other",
                    "geosite:category-ir",
                    "blogfa",
                    "bank",
                    "tebyan.net",
                    "beytoote.com",
                    "Film2movie.ws",
                    "Setare.com",
                    "downloadha.com",
                    "Sanjesh.org"
                ],
                "type": "field"
            }
        ]
    }
```
4. حتما چک کنید که مقدار tag پروتکل blackhole در بخش outbounds برابر با **blackhole** باشه مثل [این لینک](https://github.com/Gozargah/Marzban-examples/blob/0863f9c09a0d70e3f6b82670da0f4b7c881ce7d7/fully-single-port/xray_config.json#L193) 
5. با دستور های زیر داکر رو ری استارت کنید.
```sh
docker compose down
```
```sh
docker compose up -d
```

اگر از نود استفاده میکنید لازمه که [این اعمال](https://github.com/Gozargah/Marzban/discussions/38#discussioncomment-6139133) رو هم انجام بدین.

**Reply by gigil2020** [reply]
سلام
فایل env در مسیر opt/marzban موجود نیست؟
جای دیگه باید دنبالش بگردیم؟

**Reply by ghost** [reply]
سلام بصورت مخفیه. با این دستور میتونید ویرایش کنید
```sh
nano /opt/marzban/.env
```

**Reply by arp7vpn** [reply]
وقتی اینو فعال میکنی پنل دیگه بالا نمیاد!!!

**Reply by xGavazn** [reply]
https://github.com/Gozargah/Marzban/discussions/38#discussioncomment-8773374

**Reply by oldofnet** [reply]
> وقتی اینو فعال میکنی پنل دیگه بالا نمیاد!!!

آموزش رفع مسدودیت یک یا چند دامنه در این روش :

**[لینک کامنت](https://github.com/Gozargah/Marzban/discussions/38#discussioncomment-14110828)**

**Comment by lucifersanthem** [comment]
برای نود اپدیت بکنید لطفا سمت نود env و xray-config نداره که ست کنیم باتشکر

**Reply by ghost** [reply]
> برای نود اپدیت بکنید لطفا سمت نود env و xray-config نداره که ست کنیم باتشکر

برای مسدود سازی سایت های ایرانی روی سرورهای نود، این مراحل رو انجام بدین:
1. دستورات زیر رو تو محیط ترمینال سرور های نودتون بزنید. این دستورات فایل های مورد نیاز رو دانلود میکنه.
```sh
mkdir -p /var/lib/marzban/assets/
```
```sh
wget -O /var/lib/marzban/assets/geosite.dat https://github.com/v2fly/domain-list-community/releases/latest/download/dlc.dat
```
```sh
wget -O /var/lib/marzban/assets/geoip.dat https://github.com/v2fly/geoip/releases/latest/download/geoip.dat
```
```sh
wget -O /var/lib/marzban/assets/iran.dat https://github.com/bootmortis/iran-hosted-domains/releases/latest/download/iran.dat
```
2. وارد پوشه Marzban-node بشین:
```
cd ~/Marzban-node
```
3. با دستور زیر فایل docker-compose.yml رو باز کنید.

```sh
nano docker-compose.yml
```
4. محتوای فایل docker-compose.yml رو پاک کنید و مقدار زیر رو جایگزینش کنید
```
services:
  marzban-node:
    # build: .
    image: gozargah/marzban-node:latest
    restart: always
    network_mode: host

    volumes:
      - /var/lib/marzban-node:/var/lib/marzban-node
      - /var/lib/marzban/assets:/usr/local/share/xray
```
5. داکر رو ریستارک کنید
```sh
docker compose down && docker compose up -d
```
**توجه کنید که روی همه سرورهای نودتون باید این مراحل رو تکرار کنید.**

**Reply by mtohidipour** [reply]
سلام تمام این کارارو میکنم اما وقتی روی سرور مرکزی اجرا میشه سرور نود وصل نمیشه کد نود رو هم زدم

**Comment by MILADALEF** [comment]
اگر از سرور اصلی کانفیگ نداشته باشیم ، مراحلی که برای نود هست رو انجام بدیم کافیه ؟؟

**Reply by ghost** [reply]
نه باید فایل json رو سرور مرکزی رو اوکی کنی.

**Comment by esi21** [comment]
آقا توی مرزبان فایل .env رو که ادیت میکنم برای تغییر  پورت دیگه پنل بالا نمیاد
یا cert که میزنم روش دیگه بالا نمیاد

**Reply by xGavazn** [reply]
فقط مقدار زیرو به **volumes** اضافه کن همین...

`- /var/lib/marzban/assets:/usr/local/share/xray`

**Reply by nightcry1** [reply]
این کد رو باید کجا اضافه کنیم؟
تغییر پورت و اضافه کردن کد سرتیفیکیت که انجام دادیم دیگ پنل نمیاد بالا

**Comment by erffon** [comment]
سلام 
پیشنهاد میکنم این وب سایت ها رو به routing تون اضافه کنید چون پربازدید ترین وب سایت های ایرانی هستند:

"routing": {
    "domainStrategy": "IPIfNonMatch",
    "rules": [
      {
        "type": "field",
        "outboundTag": "blackhole",
        "ip": ["geoip:private", "geoip:ir"]
      },
      {
        "type": "field",
        "port": 53,
        "network": "tcp,udp",
        "outboundTag": "DNS-Internal"
      },
      {
        "type": "field",
        "outboundTag": "blackhole",
        "protocol": ["bittorrent"]
      },
      {
        "outboundTag": "blackhole",
        "domain": [
          "regexp:.*\\.ir$",
          "ext:iran.dat:ir",
          "ext:iran.dat:other",
          "geosite:category-ir",
          "blogfa",
          "bank",
          "tebyan.net",
          "beytoote.com",
          "Film2movie.ws",
          "Setare.com",
          "downloadha.com",
          "Sanjesh.org",
          "yasdl.com",
          "patoghu.com",
          "ninisite.com",
          "aparat.com",
          "mrbilit.com",
          "digikala.com",
          "varzesh3.com",
          "filimo.com",
          "torob.com",
          "telewebion.com"
        ],
        "type": "field"
      }
    ]
  }
}

**Comment by LibernetDigital** [comment]
If you plan to access domestic websites with a domestic IP, you can define a proxy in the outbounds section and use a proxy instead of using blackhole which is [explained here](https://github.com/Gozargah/Marzban/discussions/574)

**Comment by t-51** [comment]
سلام 
ببخشید یک سوال داشتیم متونیم یک استثنا قرار بدیم ؟ یعنی یک دامنه رو باز بزاریم برای مثال !

**Reply by xGavazn** [reply]
```
 "outbounds": [
      {
      "protocol": "freedom",
      "tag": "DIRECT"

    },
......

 "routing": {
    "domainStrategy": "IPIfNonMatch",    
      {
        "type": "field",
        "outboundTag": "DIRECT",
        "domain": [
          "DominA.com",
          "DominB.me",
          "DominC.dev"
        ]
      },
.........
```


برای این موضوع فقط کافیه تگی با اسم DIRECT به outbounds اضافه کنی (بالای تگ blackhole) همچنین توی قسمت routing تگی که نوشتمو اضافه کنی (بالای تگ blackhole) و در نهایت دامین‌هاتو بنویس…

**Reply by oldofnet** [reply]
آموزش رفع مسدودیت یک یا چند دامنه در این روش :

**[لینک کامنت](https://github.com/Gozargah/Marzban/discussions/38#discussioncomment-14110828)**

**Comment by MrCjxer** [comment]
سلام وقتتون بخیر

برای اینکه کلا بخوام این روتینگ رو از روی سرور چیکار باید بکنم ؟
ممنون

**Comment by mhmd-hoseinpur** [comment]
من به این روش مشکلم برطرف شد.
روی سرور نود بعد از نصب marzban-node این مراحل رو انجام دادم:
با توجه به اینکه فایل های geoip.dat و geosite.dat و iran.dat از این مسیر /usr/local/share/xray/ فراخوانی میشن
پس مسیر ها در دستورات به این مسیر تغییر دادم
این دستورات رو اجرا کنین:
```shell
mkdir -p /usr/local/share/xray
``` 
```shell
wget -O /usr/local/share/xray/geosite.dat https://github.com/v2fly/domain-list-community/releases/latest/download/dlc.dat
``` 
```shell
wget -O /usr/local/share/xray/geoip.dat https://github.com/v2fly/geoip/releases/latest/download/geoip.dat
``` 
```shell
wget -O /usr/local/share/xray/iran.dat https://github.com/bootmortis/iran-hosted-domains/releases/latest/download/iran.dat
``` 
بعد از اجرای دستورات بالا این فایل رو باید ادیت بزنین :
```shell
nano /opt/marzban-node/docker-compose.yml
``` 
و یک خط به انتها اضافه کنید:

```shell
- /var/lib/marzban/assets:/usr/local/share/xray
``` 
وارد این مسیر بشید:
```shell
cd /opt/marzban-node/
``` 
این کد رو احرا کنین که تغییرات اعمال بشه:
```shell
docker compose down && docker compose up -d
```

**Comment by morifinder** [comment]
درود بر شما وقتی دامنه ir رو ببندیم Exclude داریم که بگیم یک دامنه باز بشه مثلا پانل ما .ir هستش و اینطوری فقط با ip سرور میاد بالا اما اگر دامنه پانل بره تو وایت لیست پانل با دامنه هم میاد بالا

با تشکر

**Reply by oldofnet** [reply]
آموزش رفع مسدودیت یک یا چند دامنه در این روش :

**[لینک کامنت](https://github.com/Gozargah/Marzban/discussions/38#discussioncomment-14110828)**

**Comment by oldofnet** [comment]
اگر بعد از این کار پنل مرزبان شما باز نشد. (به احتمال زیاد به دلیل اینکه دامنه شما ir. هست) باید مراحل زیر را انجام دهید تا دامنه شما از محدودیت خارج شود و پنل شما آزاد شود.

- نکته : برای رفع مسدودیت دامنه خودتان و افزودن آن‌ها به لیست سفید (whitelist) در تنظیمات xray_config.json مرزبان، باید یک قانون جدید به بخش routing اضافه کنید که ترافیک دامنه‌های شما به جای blackhole به DIRECT هدایت شود. این کار باعث می‌شود دامنه‌های شما از مسدودسازی مستثنی شوند.

### مراحل و کد مورد نیاز برای افزودن به xray_config.json : 


**1. فایل xray_config.json را باز کنید:**
```
nano /opt/marzban/xray_config.json
```

**2. در بخش routing و قبل از قانون مربوط به blackhole برای دامنه‌های .ir، قانون زیر را اضافه کنید:**
```
{
    "type": "field",
    "outboundTag": "DIRECT",
    "domain": [
        "Domain.com",
    ]
}
```
>  بجای Domain.com دامنه پنل مرزبان را وارد کنید.


**3. همچنین، در بخش outbounds (در همان فایل xray_config.json)، مطمئن شوید که یک outbound با تگ DIRECT وجود دارد. اگر وجود ندارد، آن را اضافه کنید:**
```
{
    "protocol": "freedom",
    "tag": "DIRECT"
}
```

### نمونه کد نهایی : 
```
"routing": {
    "domainStrategy": "IPIfNonMatch",
    "rules": [
        {
            "type": "field",
            "outboundTag": "DIRECT",
            "domain": [
                "Domain.com",
            ]
        },
        {
            "type": "field",
            "outboundTag": "blackhole",
            "ip": [
                "geoip:private",
                "geoip:ir"
            ]
        },
        {
            "type": "field",
            "port": 53,
            "network": "tcp,udp",
            "outboundTag": "DNS-Internal"
        },
        {
            "type": "field",
            "outboundTag": "blackhole",
            "protocol": [
                "bittorrent"
            ]
        },
        {
            "outboundTag": "blackhole",
            "domain": [
                "regexp:.*\\.ir$",
                "ext:iran.dat:ir",
                "ext:iran.dat:other",
                "geosite:category-ir",
                "blogfa",
                "bank",
                "tebyan.net",
                "beytoote.com",
                "Film2movie.ws",
                "Setare.com",
                "downloadha.com",
                "Sanjesh.org"
            ],
            "type": "field"
        }
    ]
}
```
>  بجای Domain.com دامنه پنل مرزبان را وارد کنید



### بعد از اعمال تغییرات برای بررسی مراحل زیر را انجام دهید :

**1. مرزبان را ری‌استارت کرده :**
`marzban restart`
**2. درصورت مشکل Docker را ری‌استارت کنید.**

Training by [@oldofnet](https://github.com/oldofnet), made with ❤️

**Reply by LixhKing** [reply]
👌

**Comment by Sooshka-X** [comment]
سلام
آیا امکانش هست که سایتهای ایرانی بلاک نشن و از اینترنت خود کاربر استفاده بشه؟
یعنی Splittunnel انجام بشه و سایتهای غیر از ایران از پروکسی عبور کنن؟
تشکر

**Comment by ReMlnDeR** [comment]
تمام مراحل رو میرم و بازم سرور نود به مرکزی وصل نمیشه
